internal/pkg/config/app.go

```diff
@@ -74,8 +76,18 @@ func (s *Store) UpdateApplication(application *Application) error {
 
 // GetApplication fetches an application by name. If it can't be found, return a ErrNoSuchApplication
 func (s *Store) GetApplication(applicationName string) (*Application, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
```

Comment:
My understanding from the error that we get:
```
✘ get application demo configuration: get application demo: NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: RequestError: send request failed
caused by: Get "http://169.254.169.254/latest/meta-data/iam/security-credentials/": dial tcp 169.254.169.254:80: connect: host is down
/Users/karakuse/Development/copilot-cli/bin/local/copilot svc init  0.02s user 0.01s system 0% cpu 5.555 total
```

The step that takes the longest time is while it's trying to see if it can fetch EC2 IAM role credentials 💭 

If I comment our custom timeouts:
https://github.com/aws/copilot-cli/blob/0bea71053c1f260ad4f832d7cb550353941f5d51/internal/pkg/aws/sessions/sessions.go#L26-L28

Then we receive an error much faster. Whereas now it takes 8*30s = 4 mins.
Maybe, we can change our defaults or create a test "default" session with very short timeouts (2 retries, 3 sec timeout) and see if we get an error.